pxtwinrt/hiddeploy.ts

```diff
@@ -49,44 +53,78 @@ namespace pxt.winrt {
-                                        const values: number[] = [];
-                                        while (dr.unconsumedBufferLength) {
-                                            values.push(dr.readByte());
-                                        }
-                                        if (values.length == 65 && values[0] === 0) {
-                                            values.shift()
-                                        }
-                                        this.onData(new Uint8Array(values));
-                                    });
-                                    return Promise.resolve();
-                                }
-                                pxt.debug("no hid device found");
-                                return Promise.reject(new Error("no hid device found"));
-                            });
+                    pxt.debug(`hid connect to ${device.name} (${device.id})`);
+                    return whid.fromIdAsync(device.id, Windows.Storage.FileAccessMode.readWrite);
+                })
+                .then((r: WHID) => {
+                    this.dev = r;
+                    if (!this.dev) {
+                        pxt.debug("can't connect to hid device");
+                        return Promise.reject(new Error("can't connect to hid device"));
                     }
-                    return Promise.reject(new Error("no hid device found"));
-                }))
+                    pxt.debug(`hid device version ${this.dev.version}`);
+                    this.dev.addEventListener("inputreportreceived", (e) => {
+                        pxt.debug(`input report`)
+                        const dr = Windows.Storage.Streams.DataReader.fromBuffer(e.report.data);
+                        const values: number[] = [];
+                        while (dr.unconsumedBufferLength) {
+                            values.push(dr.readByte());
+                        }
+                        if (values.length == 65 && values[0] === 0) {
+                            values.shift()
+                        }
+                        this.onData(new Uint8Array(values));
+                    });
+                    return Promise.resolve();
+                })
                 .catch((e) => {
                     const err = new Error(U.lf("Device not found"));
                     (<any>err).notifyUser = true;
                     return Promise.reject(err);
                 });
         }
+
+        private initUf2Selectors() {
+            const whid = Windows.Devices.HumanInterfaceDevice.HidDevice;
+            // Selectors common to all UF2 bootloaders
+            WindowsRuntimeIO.uf2Selectors = [
+                // UF2 Bootloader v1.23.0
+                whid.getDeviceSelector(0xFF97, 0x0001, 0x239A, 0x0018),
+                // UF2 Bootloader v1.22.0
+                whid.getDeviceSelector(0xFF00, 0x0001, 0x239A, 0x0019)
```

Comment:
These 2 specific IDs are for the UF2 bootloader, which is target-independent (any target using UF2 will use these IDs)